op-node/rollup/derive/l1_block_info.go

```diff
@@ -423,3 +427,34 @@ func DepositsCompleteBytes(seqNumber uint64, l1Info eth.BlockInfo) ([]byte, erro
 	}
 	return opaqueDepositsCompleteTx, nil
 }
+
+func AddDependencyDeposit(seqNumber uint64, block eth.BlockInfo, newDependency *big.Int) (*types.DepositTx, error) {
```

Comment:
While the other functions don't have documentation, since we do not have specs for this it would be good to add something quick

```
// AddDependencyDeposit creates a system transaction to indicate that
// a new dependency was added to the cluster through a network upgrade
// as specified in the ClusterConfig
```